src/envoy/http/authn/authenticator_base.cc

```diff
@@ -29,6 +29,24 @@ namespace Http {
 namespace Istio {
 namespace AuthN {
 
+namespace {
+// The default header name for an exchanged token
+static const std::string kExchangedTokenHeaderName = "ingress-authorization";
+
+// Return whether the header for an exchanged token is found
+bool FindHeaderOfExchangedToken(const iaapi::Jwt& jwt) {
+  bool found = false;
+  for (auto h : jwt.jwt_headers()) {
```

Comment:
If we make assumption on the location value, should we also make assumption/requirement that the Jwt spec should declare only one header location. It seems weird (and could be wrong) to define more than one for aptoken.
+ @liminw

Reply:
Please change this per agreement. You can add check for  jwt_headers() size and compare the only one in there.